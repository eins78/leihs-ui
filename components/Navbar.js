import f from 'lodash'
import React, { Fragment as F } from 'react'
// import cx from 'classnames'
import { ColorTint } from '../lib/color-fns'
import {
  // Badge,
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItemLink,
  // NavItemAnchor,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
  // DropdownItemLink
} from './Bootstrap'

import Icon from './Icons'
// import { DisplayName } from './decorators'

const isDev = process.env.NODE_ENV === 'development'
export const BASE_COLOR = '#563d7c' // bootstrap docs purple
// export const BASE_COLOR = '#343a40' // bootstrap bg-dark

const Brand = ({ title }) => (
  <F>
    <Icon.LeihsLogo className="mr-2" />
    {title}
  </F>
)

export default class MainNav extends React.Component {
  state = {
    isOpen: false
  }
  toggleOpen() {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }
  render({ props, state } = this) {
    const { me, appTitle, appMenu, appColor } = props.config
    const bgColor = ColorTint(BASE_COLOR, appColor)

    return (
      <Navbar
        dark
        color="dark"
        expand="lg"
        className="navbar-leihs"
        style={{ backgroundColor: `${bgColor} !important` }}
      >
        <NavbarBrand exact to="/">
          <Brand title={appTitle} />
        </NavbarBrand>

        <NavbarToggler onClick={e => this.toggleOpen()} />

        <Collapse isOpen={state.isOpen} navbar>
          <Nav className="mr-auto" navbar>
            {f.map(appMenu, ({ title, href, icon, active, submenu, attr }) => {
              const IconEl = Icon[icon]
              return (
                <NavItemLink href={href} active={active} {...attr}>
                  {IconEl ? <IconEl fixedWidth spaced /> : false} {title}
                </NavItemLink>
              )
            })}
          </Nav>

          <Nav className="ml-auto" navbar>
            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                <Icon.LeihsProcurement />
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem href="/borrow">
                  <Icon.LeihsBorrow /> Ausleihen
                </DropdownItem>
                <DropdownItem divider />
                <DropdownItem href="/admin">
                  <Icon.LeihsAdmin /> Admin
                </DropdownItem>
                <DropdownItem divider />
                <DropdownItem href="/procure">
                  <Icon.LeihsProcurement /> Bedarfsermittlung
                </DropdownItem>
                <DropdownItem divider />
                <DropdownItem href="/manage">
                  <Icon.LeihsManage /> Manage
                </DropdownItem>
                <DropdownItem divider />
                {isDev && (
                  <DropdownItem href="/">
                    <Icon.LeihsStyleguide /> Styleguide
                  </DropdownItem>
                )}
              </DropdownMenu>
            </UncontrolledDropdown>

            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                <Icon.User size="lg" />
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem tag="span">
                  <form action="/sign-out" method="POST">
                    <button type="submit">Ausloggen</button>
                  </form>
                </DropdownItem>
                <DropdownItem>{tmpUserInfo(me.user)}</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>

            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                <Icon.Language />
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem>[TODO]</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
          </Nav>
        </Collapse>
      </Navbar>
    )
  }
}

const tmpUserInfo = user => (
  <F>
    <pre>
      <small>{JSON.stringify(user, 0, 2)}</small>
    </pre>
  </F>
)

const Let = ({ children, ...props }) => children(props)
