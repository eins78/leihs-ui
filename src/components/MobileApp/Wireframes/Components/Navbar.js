import React from 'react'

export default function Navbar({ brandName = 'Leihs' }) {
  return <nav
    className="tmp-navbar ui-main-nav navbar navbar-light text-xl py-0 px-4 sticky-top flex-nowrap justify-content-between"
  >
    <div className="navbar-nav w-100">
      <div className="mr-auto">
        <a className="nav-item nav-link px-0">
          <span className="ui-icon ui-menu-icon">â˜°</span>
        </a>
      </div>
    </div>
    <div className="mx-auto">
      {!!brandName && (
        <a className="navbar-brand m-0 font-semibold text-xl">
          {' ' + brandName + ' '}
        </a>
      )}
    </div>
    <div className="navbar-nav w-100">
      <div className="ml-auto d-flex align-items-center">
        <div className="mx-auto px-2 text-xs">
          <span className="text-color-info" />
        </div>{' '}
        <a href="/app/borrow/order" className="nav-item nav-link px-0">
          (3)
        </a>
      </div>
    </div>
  </nav>
}