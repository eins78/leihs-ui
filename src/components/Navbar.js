import f from 'lodash'
import React, { Fragment as F } from 'react'
import cx from 'classnames'
import { ColorTint } from '../lib/color-fns'
import {
  // Badge,
  Collapse,
  Navbar as BsNavbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItemLink,
  // NavItemAnchor,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
  // DropdownItemLink
} from './Bootstrap'

import Icon from './Icons'
import { NavbarLogin } from './SignInUI'

// export const BASE_COLOR = '#563d7c' // bootstrap docs purple
export const BASE_COLOR = '#343a40' // bootstrap bg-dark
// const LEIHS_GREEN = '#afec81'

const defaults = {
  homeUrl: '/'
}

const Brand = ({ title }) => (
  <F>
    <Icon.LeihsLogo className="mr-2" />
    {title}
  </F>
)

export default class Navbar extends React.Component {
  static defaultProps = {
    config: {},
    bgBaseColor: BASE_COLOR
  }

  state = {
    isOpen: false
  }
  toggleOpen() {
    this.setState({
      isOpen: !this.state.isOpen
    })
  }

  render({ props, state } = this) {
    const { bgBaseColor, config } = props
    const {
      me,
      appTitle,
      appMenu,
      appColor,
      subApps,
      returnTo,
      csrfToken
    } = config
    const { homeUrl } = defaults

    const bgColor =
      props.bgColor || appColor ? ColorTint(bgBaseColor, appColor) : bgBaseColor

    return (
      <BsNavbar
        dark
        color="dark"
        expand="lg"
        className={cx('navbar-leihs', props.className)}
        style={{ backgroundColor: `${bgColor} !important` }}
      >
        <NavbarBrand href={homeUrl}>
          <Brand title={appTitle} />
        </NavbarBrand>

        <NavbarToggler onClick={e => this.toggleOpen()} />

        <Collapse isOpen={state.isOpen} navbar>
          <Nav className="mr-auto" navbar>
            {f.map(
              appMenu,
              ({ title, href, icon, active, submenu, attr }, i) => {
                const IconEl = Icon[icon]
                return (
                  <NavItemLink key={i} href={href} active={active} {...attr}>
                    {IconEl ? <IconEl fixedWidth spaced /> : false} {title}
                  </NavItemLink>
                )
              }
            )}
          </Nav>

          <Nav className="ml-auto" navbar>
            <SubAppDropdown subApps={subApps} />

            {f.isEmpty(me) || f.isEmpty(me.user) ? (
              <NavbarLogin returnTo={returnTo} />
            ) : (
              <UncontrolledDropdown nav inNavbar>
                <DropdownToggle nav caret>
                  <Icon.User size="lg" />
                </DropdownToggle>
                <DropdownMenu right>
                  <DropdownItem tag="span">
                    <form action="/sign-out" method="POST">
                      {!!csrfToken && (
                        <input
                          type="hidden"
                          name="csrf-token"
                          value={csrfToken}
                        />
                      )}
                      <button type="submit">Ausloggen</button>
                    </form>
                  </DropdownItem>
                  <DropdownItem>{tmpUserInfo({ me })}</DropdownItem>
                </DropdownMenu>
              </UncontrolledDropdown>
            )}

            <UncontrolledDropdown nav inNavbar>
              <DropdownToggle nav caret>
                <Icon.Language />
              </DropdownToggle>
              <DropdownMenu right>
                <DropdownItem>[TODO]</DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
          </Nav>
        </Collapse>
      </BsNavbar>
    )
  }
}

const tmpUserInfo = user => (
  <F>
    <pre>
      <small>{JSON.stringify(user, 0, 2)}</small>
    </pre>
  </F>
)

const SubAppDropdown = ({ subApps }) =>
  f.some(subApps) && (
    <UncontrolledDropdown nav inNavbar>
      <DropdownToggle nav caret>
        <Icon.LeihsProcurement />
      </DropdownToggle>
      <DropdownMenu right>
        {f.map(
          f.keys(f.fromPairs(f.filter(f.toPairs(subApps), '1'))),
          (subApp, i, a) => {
            const withDivider = i + 1 < a.length // not if last
            let item

            if (subApp === 'borrow')
              item = (
                <DropdownItem href="/borrow">
                  <Icon.LeihsBorrow /> Ausleihen
                </DropdownItem>
              )

            if (subApp === 'admin')
              item = (
                <DropdownItem href="/admin">
                  <Icon.LeihsAdmin /> Admin
                </DropdownItem>
              )

            if (subApp === 'procure')
              item = (
                <DropdownItem href="/procure">
                  <Icon.LeihsProcurement /> Bedarfsermittlung
                </DropdownItem>
              )

            if (subApp === 'manage')
              item = (
                <DropdownItem href="/manage">
                  <Icon.LeihsManage /> Manage
                </DropdownItem>
              )

            if (subApp === 'styleguide')
              item = (
                <DropdownItem href="/">
                  <Icon.LeihsStyleguide /> Styleguide
                </DropdownItem>
              )

            return (
              <F key={subApp}>
                {item}
                {withDivider && <DropdownItem divider />}
              </F>
            )
          }
        )}
      </DropdownMenu>
    </UncontrolledDropdown>
  )

// const Let = ({ children, ...props }) => children(props)
