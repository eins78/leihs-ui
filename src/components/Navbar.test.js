import React from 'react'
import renderer from 'react-test-renderer'

import Navbar from './Navbar'
const LEIHS_GREEN = '#afec81'

const examples = [
  {
    name: 'base',
    content: <Navbar />
  },
  {
    name: 'on root page, logged out',
    content: (
      <Navbar
        config={{
          me: false,
          appTitle: 'leihs',
          appMenu: null,
          subApps: false,
          appColor: null,
          returnTo: '/someapp'
        }}
      />
    )
  },
  {
    name: 'no user, marks default lang',
    content: (
      <Navbar
        config={{
          me: false,
          locales: [
            {
              id: 1,
              locale_name: 'en',
              name: 'EN',
              isDefault: false,
              isSelected: false
            },
            {
              id: 2,
              locale_name: 'de',
              name: 'DE',
              isDefault: true,
              isSelected: false
            }
          ]
        }}
      />
    )
  },
  {
    name: 'no user, with selected lang',
    content: (
      <Navbar
        config={{
          me: false,
          locales: [
            {
              id: 1,
              locale_name: 'en',
              name: 'EN',
              isDefault: false,
              isSelected: true
            },
            {
              id: 2,
              locale_name: 'de',
              name: 'DE',
              isDefault: true,
              isSelected: false
            }
          ]
        }}
      />
    )
  },
  {
    name: 'no user, with selected lang same as default lang',
    content: (
      <Navbar
        config={{
          me: false,
          locales: [
            {
              id: 1,
              locale_name: 'en',
              name: 'EN',
              isDefault: false,
              isSelected: false
            },
            {
              id: 2,
              locale_name: 'de',
              name: 'DE',
              isDefault: true,
              isSelected: true
            }
          ]
        }}
      />
    )
  },
  {
    name: 'user without lang setting, marks default lang',
    content: (
      <Navbar
        config={{
          me: { user: { login: 'ME' } },
          locales: [
            {
              id: 1,
              locale_name: 'en',
              name: 'EN',
              isDefault: false,
              isSelected: false
            },
            {
              id: 2,
              locale_name: 'de',
              name: 'DE',
              isDefault: true,
              isSelected: false
            }
          ]
        }}
      />
    )
  },
  {
    name: 'user with lang setting, marks selected lang',
    content: (
      <Navbar
        config={{
          me: { user: { login: 'ME' } },
          locales: [
            {
              id: 1,
              locale_name: 'en',
              name: 'EN',
              isDefault: false,
              isSelected: true
            },
            {
              id: 2,
              locale_name: 'de',
              name: 'DE',
              isDefault: true,
              isSelected: false
            }
          ]
        }}
      />
    )
  },

  {
    name: 'usage in shared app',
    content: (
      <Navbar
        config={{
          appTitle: 'leihs',
          appColor: LEIHS_GREEN,
          appMenu: [
            {
              title: 'one',
              href: '/1',
              icon: 'Settings',
              active: false,
              attr: { 'data-foo': 'bar' }
            },
            {
              title: 'two',
              href: '/2',
              icon: 'Categories',
              active: true
            }
          ],
          me: {
            user: {
              name: 'Normin Normalo'
            }
          }
        }}
      />
    )
  }
]

examples.forEach(({ name, content }) => {
  it(`${name}: renders correctly`, () => {
    expect(renderer.create(content).toJSON()).toMatchSnapshot()
  })
})
