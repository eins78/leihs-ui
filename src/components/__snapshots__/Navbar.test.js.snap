// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base: renders correctly 1`] = `
<nav
  className="navbar-leihs navbar navbar-expand-lg navbar-dark bg-dark"
  style={
    Object {
      "backgroundColor": "#343a40 !important",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-dolly fa-w-18 mr-2"
      data-icon="dolly"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.2 277.7c18 5 34.7 13.4 49.5 24.7l161.5-53.8c8.4-2.8 12.9-11.9 10.1-20.2L454.9 47.2c-2.8-8.4-11.9-12.9-20.2-10.1l-61.1 20.4 33.1 99.4L346 177l-33.1-99.4-61.6 20.5c-8.4 2.8-12.9 11.9-10.1 20.2l53 159.4zm281 48.7L565 296c-2.8-8.4-11.9-12.9-20.2-10.1l-213.5 71.2c-17.2-22-43.6-36.4-73.5-37L158.4 21.9C154 8.8 141.8 0 128 0H16C7.2 0 0 7.2 0 16v32c0 8.8 7.2 16 16 16h88.9l92.2 276.7c-26.1 20.4-41.7 53.6-36 90.5 6.1 39.4 37.9 72.3 77.3 79.2 60.2 10.7 112.3-34.8 113.4-92.6l213.3-71.2c8.3-2.8 12.9-11.8 10.1-20.2zM256 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </a>
  <button
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="mr-auto navbar-nav"
    />
    <ul
      className="ml-auto navbar-nav"
    >
      <form
        action="/sign-in"
        className="form-inline my-2 my-lg-0"
        method="POST"
      >
        <div
          className="input-group"
        >
          <input
            aria-describedby="button-addon2"
            aria-label="Nutzername/Email"
            className="form-control"
            name="user"
            placeholder="Nutzername/Email"
            type="text"
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-success"
              id="button-addon2"
              type="submit"
            >
              Login
            </button>
          </div>
        </div>
      </form>
      <li
        className="dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          href="#"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <div
          aria-hidden={true}
          className="dropdown-menu dropdown-menu-right"
          role="menu"
          tabIndex="-1"
        >
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            [TODO]
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`on root page, logged out: renders correctly 1`] = `
<nav
  className="navbar-leihs navbar navbar-expand-lg navbar-dark bg-dark"
  style={
    Object {
      "backgroundColor": "#343a40 !important",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-dolly fa-w-18 mr-2"
      data-icon="dolly"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.2 277.7c18 5 34.7 13.4 49.5 24.7l161.5-53.8c8.4-2.8 12.9-11.9 10.1-20.2L454.9 47.2c-2.8-8.4-11.9-12.9-20.2-10.1l-61.1 20.4 33.1 99.4L346 177l-33.1-99.4-61.6 20.5c-8.4 2.8-12.9 11.9-10.1 20.2l53 159.4zm281 48.7L565 296c-2.8-8.4-11.9-12.9-20.2-10.1l-213.5 71.2c-17.2-22-43.6-36.4-73.5-37L158.4 21.9C154 8.8 141.8 0 128 0H16C7.2 0 0 7.2 0 16v32c0 8.8 7.2 16 16 16h88.9l92.2 276.7c-26.1 20.4-41.7 53.6-36 90.5 6.1 39.4 37.9 72.3 77.3 79.2 60.2 10.7 112.3-34.8 113.4-92.6l213.3-71.2c8.3-2.8 12.9-11.8 10.1-20.2zM256 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Leihs
  </a>
  <button
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="mr-auto navbar-nav"
    />
    <ul
      className="ml-auto navbar-nav"
    >
      <form
        action="/sign-in"
        className="form-inline my-2 my-lg-0"
        method="POST"
      >
        <div
          className="input-group"
        >
          <input
            aria-describedby="button-addon2"
            aria-label="Nutzername/Email"
            className="form-control"
            name="user"
            placeholder="Nutzername/Email"
            type="text"
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-success"
              id="button-addon2"
              type="submit"
            >
              Login
            </button>
          </div>
          <input
            name="return_to"
            type="hidden"
            value="/someapp"
          />
        </div>
      </form>
      <li
        className="dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          href="#"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <div
          aria-hidden={true}
          className="dropdown-menu dropdown-menu-right"
          role="menu"
          tabIndex="-1"
        >
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            [TODO]
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`usage in shared app: renders correctly 1`] = `
<nav
  className="navbar-leihs navbar navbar-expand-lg navbar-dark bg-dark"
  style={
    Object {
      "backgroundColor": "hsl(94.19999999999999, 100%, 12.9%) !important",
    }
  }
>
  <a
    className="navbar-brand"
    href="/"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-dolly fa-w-18 mr-2"
      data-icon="dolly"
      data-prefix="fas"
      role="img"
      style={Object {}}
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.2 277.7c18 5 34.7 13.4 49.5 24.7l161.5-53.8c8.4-2.8 12.9-11.9 10.1-20.2L454.9 47.2c-2.8-8.4-11.9-12.9-20.2-10.1l-61.1 20.4 33.1 99.4L346 177l-33.1-99.4-61.6 20.5c-8.4 2.8-12.9 11.9-10.1 20.2l53 159.4zm281 48.7L565 296c-2.8-8.4-11.9-12.9-20.2-10.1l-213.5 71.2c-17.2-22-43.6-36.4-73.5-37L158.4 21.9C154 8.8 141.8 0 128 0H16C7.2 0 0 7.2 0 16v32c0 8.8 7.2 16 16 16h88.9l92.2 276.7c-26.1 20.4-41.7 53.6-36 90.5 6.1 39.4 37.9 72.3 77.3 79.2 60.2 10.7 112.3-34.8 113.4-92.6l213.3-71.2c8.3-2.8 12.9-11.8 10.1-20.2zM256 464c-26.5 0-48-21.5-48-48s21.5-48 48-48 48 21.5 48 48-21.5 48-48 48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Leihs
  </a>
  <button
    className="navbar-toggler"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    style={Object {}}
  >
    <ul
      className="mr-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          data-foo="bar"
          href="/1"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-cog fa-w-16 fa-fw mr-1"
            data-icon="cog"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          one
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          href="/2"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sitemap fa-w-20 fa-fw mr-1"
            data-icon="sitemap"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M128 352H32c-17.67 0-32 14.33-32 32v96c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32v-96c0-17.67-14.33-32-32-32zm-24-80h192v48h48v-48h192v48h48v-57.59c0-21.17-17.23-38.41-38.41-38.41H344v-64h40c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32H256c-17.67 0-32 14.33-32 32v96c0 17.67 14.33 32 32 32h40v64H94.41C73.23 224 56 241.23 56 262.41V320h48v-48zm264 80h-96c-17.67 0-32 14.33-32 32v96c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32v-96c0-17.67-14.33-32-32-32zm240 0h-96c-17.67 0-32 14.33-32 32v96c0 17.67 14.33 32 32 32h96c17.67 0 32-14.33 32-32v-96c0-17.67-14.33-32-32-32z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          two
        </a>
      </li>
    </ul>
    <ul
      className="ml-auto navbar-nav"
    >
      <li
        className="dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          href="#"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user-circle fa-w-16 fa-lg "
            data-icon="user-circle"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <div
          aria-hidden={true}
          className="dropdown-menu dropdown-menu-right"
          role="menu"
          tabIndex="-1"
        >
          <span
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
          >
            <form
              action="/sign-out"
              method="POST"
            >
              <button
                type="submit"
              >
                Ausloggen
              </button>
            </form>
          </span>
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <pre>
              <small>
                {
  "me": {
    "user": {
      "name": "Normin Normalo"
    }
  }
}
              </small>
            </pre>
          </button>
        </div>
      </li>
      <li
        className="dropdown nav-item"
        onKeyDown={[Function]}
      >
        <a
          aria-expanded={false}
          aria-haspopup={true}
          className="dropdown-toggle nav-link"
          href="#"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
        <div
          aria-hidden={true}
          className="dropdown-menu dropdown-menu-right"
          role="menu"
          tabIndex="-1"
        >
          <button
            className="dropdown-item"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            [TODO]
          </button>
        </div>
      </li>
    </ul>
  </div>
</nav>
`;
